src/dmd/backend/blockopt.d

```diff
@@ -6,80 +6,111 @@
-#include        "oper.h"
-#include        "el.h"
-#include        "type.h"
-#include        "global.h"
-#include        "go.h"
-#include        "code.h"
-#if SCPP
-#include        "parser.h"
-#include        "iasm.h"
-#endif
-
-static char __file__[] = __FILE__;      /* for tassert.h                */
-#include        "tassert.h"
-
-STATIC void bropt(void);
-STATIC void brrear(void);
-STATIC void search(block *b);
-STATIC void elimblks(void);
-STATIC int  mergeblks(void);
-STATIC void blident(void);
-STATIC void blreturn(void);
-STATIC void brmin(void);
-STATIC void bltailmerge(void);
-STATIC void block_check();
-STATIC void brtailrecursion();
-STATIC elem * assignparams(elem **pe,int *psi,elem **pe2);
-STATIC void emptyloops();
-STATIC int el_anyframeptr(elem *e);
-STATIC void blassertsplit();
-
-unsigned numblks;       // number of basic blocks in current function
+version (SPP) {} else
+{
+
+import core.stdc.stdio;
+import core.stdc.string;
+import core.stdc.time;
+import core.stdc.stdlib;
+
+import dmd.backend.cc;
+import dmd.backend.cdef;
+import dmd.backend.oper;
+import dmd.backend.dlist;
+import dmd.backend.dvec;
+import dmd.backend.el;
+import dmd.backend.type;
+import dmd.backend.global;
+import dmd.backend.goh;
+import dmd.backend.code;
+import dmd.backend.ty;
+version (SCPP)
+{
+import parser;
+import iasm;
+}
+
+
+version (SCPP)
+    enum SCPP_OR_NTEXCEPTIONS = true;
+else version (NTEXCEPTIONS)
+    enum SCPP_OR_NTEXCEPTIONS = true;
+else
+    enum SCPP_OR_NTEXCEPTIONS = false;
+
+extern(C++):
+
+extern(C) void *mem_fcalloc(size_t numbytes); // tk/mem.c
+extern(C) void *mem_free(size_t numbytes); // tk/mem.c
+extern(C) void *mem_ffree(void* p) // tk/mem.h
+{
+    version (MEM_NONE)
+    {
+        return null;
+    }
+    else version (MEM_DEBUG)
+    {
+        return mem_free(p);
+    }
+    else
+    {
+        return null;
+    }
+}
+version(Posix)
+{
+    extern(C) void *mem_realloc(void* p, size_t u); // tk/mem.c
+    extern(C) void *util_realloc(void *oldp,uint n,uint size) // util2.c
+    {
+        return mem_realloc(oldp, n * size);
+    }
+}
+else
+{
+    void *util_realloc(void *oldp,uint n,uint size); // util2.c
+}
+
+private __gshared
+{
+
+uint numblks;       // number of basic blocks in current function
 block *startblock;      /* beginning block of function                  */
                         /* (can have no predecessors)                   */
 
-block **dfo = NULL;     /* array of depth first order                   */
-unsigned dfotop;        /* # of items in dfo[]                          */
+block **dfo = null;     /* array of depth first order                   */
+uint dfotop;        /* # of items in dfo[]                          */
 
 block *curblock;        /* current block being read in                  */
 block *block_last;      // last block read in
 
-static block * block_freelist;
+block * block_freelist;
 
 ////////////////////////////
 // Storage allocator.
 
-static block blkzero;
```

Comment:
Walter usually coverts `static` to `private`.

Reply:
This is in a `private __gshared` block.

Reply:
Ok.